scripts/freeze-repos.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+set -euC
+
+function update_repository_version {
+    local -r repos_file="$1"
+    local -r repository="$2"
+
+    url=$(yq ".repositories[] | select(key == \"$repository\").url" "$repos_file")
+    version=$(yq ".repositories[] | select(key == \"$repository\").version" "$repos_file")
+
+    tag_hash=$(git ls-remote --tags "$url" "$version" | awk '{print $1}')
+    branch_hash=$(git ls-remote --heads "$url" "$version" | awk '{print $1}')
+
+    if [ -n "$tag_hash" ]; then
+        new_version="$tag_hash"
+    elif [ -n "$branch_hash" ]; then
+        new_version="$branch_hash"
+    else
+        new_version="$version"
+    fi
+
+    echo "$repository: $new_version"
+    yq --inplace ".repositories.\"$repository\".version = \"$new_version\"" "$repos_file"
```

Comment:
Looks like there were change in command syntax in newer version https://github.com/mikefarah/yq/issues/1203.
Same change would be necessary for other lines where 'yq' is used.

```suggestion
    yq e --inplace ".repositories.\"$repository\".version = \"$new_version\"" "$repos_file"
```

Reply:
I guess it's for older versions of `yq`.
https://github.com/mikefarah/yq/releases/tag/v4.18.1

> eval is now the default command, you can leave it out dancers https://github.com/mikefarah/yq/issues/113